bundle/config/validate/schema_references.go

```diff
@@ -0,0 +1,136 @@
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Unity Catalog pipeline should have a schema or target defined",
+				Detail: `The target or schema field is required for UC pipelines. Reason: DLT
+requires specifying a target schema for UC pipelines. Please use the
+TEMPORARY keyword in the CREATE MATERIALIZED VIEW or CREATE STREAMING
+TABLE statement if you do not wish to publish your dataset.`,
+				Locations: rb.Config().GetLocations("resources.pipelines." + k),
+				Paths: []dyn.Path{
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("schema")),
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("target")),
+				},
+			})
+			continue
+		case p.Schema != "" && p.Target != "":
+			locations := rb.Config().GetLocations("resources.pipelines." + k + ".schema")
+			locations = append(locations, rb.Config().GetLocations("resources.pipelines."+k+".target")...)
+
+			// The Databricks Terraform provider already has client side validation
+			// that does not allow this today. Having this here allows us to float
+			// this validation on `bundle validate` and provide location information.
+			diags = append(diags, diag.Diagnostic{
+				Severity:  diag.Error,
+				Summary:   "Both schema and target are defined in a Unity Catalog pipeline. Only one of them should be defined.",
+				Locations: locations,
+				Paths: []dyn.Path{
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("schema")),
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("target")),
+				},
+			})
+			continue
+		case p.Schema != "":
+			schemaName = p.Schema
+			fieldPath = dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("schema"))
+			schemaLocation = rb.Config().GetLocations("resources.pipelines." + k + ".schema")
+		case p.Target != "":
+			schemaName = p.Target
+			fieldPath = dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("target"))
+			schemaLocation = rb.Config().GetLocations("resources.pipelines." + k + ".target")
+		}
+
+		// Check if the schema is defined in the bundle.
+		matchLocations, matchPath, found := findSchemaInBundle(rb, p.Catalog, schemaName)
+		if !found {
+			continue
+		}
+
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  fmt.Sprintf("Use ${%s.name} syntax to refer to the UC schema instead of directly using its name %q", matchPath, schemaName),
+			Detail: fmt.Sprintf(`Using ${%s.name} will allow DABs to capture the deploy time dependency this DLT pipeline
+has on the schema %q and deploy changes to the schema before deploying the pipeline.`, matchPath, schemaName),
+			Locations: append(schemaLocation, matchLocations...),
+			Paths: []dyn.Path{
+				fieldPath,
+				matchPath,
+			},
+		})
+	}
+
+	for k, v := range rb.Config().Resources.Volumes {
```

Comment:
Please break the body of these loops into a separate function to keep it small(-er).

Reply:
done